crates/topos-tce/src/app_context.rs

```diff
@@ -267,315 +262,319 @@ impl AppContext {
+                //     .iter()
+                //     .map(|peer_id| {
+                //         debug!("Peer {} is sending Echo to {}", &my_peer_id, &peer_id);
+                //         self.network_client
+                //             .send_request::<_, NetworkMessage>(
+                //                 *peer_id,
+                //                 data.clone(),
+                //                 RetryPolicy::N(3),
+                //             )
+                //             .instrument(span.clone())
+                //             .with_context(span.context())
+                //     })
+                //     .collect();
+                //
+                // spawn(async move {
+                //     let _results: Vec<_> = future_pool
+                //         .collect()
+                //         .instrument(span.clone())
+                //         .with_context(span.context())
+                //         .await;
+                // });
             }
 
             ProtocolEvents::Ready {
-                peers,
                 certificate_id,
                 ctx,
             } => {
                 let span = info_span!(
-                    "SEND Ready",
+                    parent: &ctx,
+                    "SEND Outbound Ready",
                     peer_id = self.network_client.local_peer_id.to_string(),
                     "otel.kind" = "producer",
                 );
-                span.set_parent(ctx);
                 let my_peer_id = self.network_client.local_peer_id;
-                let data: Vec<u8> = NetworkMessage::from(TceCommands::OnReady {
-                    from_peer: self.network_client.local_peer_id,
+                let data = NetworkMessage::from(TceCommands::OnReady {
                     certificate_id,
                     ctx: PropagationContext::inject(&span.context()),
-                })
-                .into();
-
-                let future_pool = peers
-                    .iter()
-                    .map(|peer_id| {
-                        debug!("Peer {} is sending Ready to {}", &my_peer_id, &peer_id);
-                        self.network_client
-                            .send_request::<_, NetworkMessage>(
-                                *peer_id,
-                                data.clone(),
-                                RetryPolicy::N(3),
-                            )
-                            .instrument(span.clone())
-                            .with_context(span.context())
-                    })
-                    .collect::<Vec<_>>();
-
-                spawn(async move {
-                    let _results = join_all(future_pool)
-                        .instrument(span.clone())
-                        .with_context(span.context().clone())
-                        .await;
                 });
+
+                if let Err(e) = self
+                    .network_client
+                    .publish::<NetworkMessage>(topos_p2p::TOPOS_ECHO, data)
```

Comment:
```suggestion
                    .publish::<NetworkMessage>(topos_p2p::TOPOS_READY, data)
```

looks like that brings an infinite loop of messages as from the doc the `message_id` is by default `( peer_id | sequence_number )` of the publisher, so it seems to me that they will be taken as different payloads if I'm not mistaken

btw we may want then to switch the default [message_id()] implementation (https://docs.rs/gossipsub/latest/gossipsub/struct.GossipsubConfig.html#method.message_id) from the concatenation to the hash of the message instead no?